packages/sku/src/services/packageManager/pnpmConfig.ts

```diff
@@ -0,0 +1,156 @@
+      ),
+    );
+
+    return;
+  }
+
+  // There is potentially conflicting config somewhere, so log a message instructing
+  // the user to update the config manually
+  if (action === 'log') {
+    console.log(
+      chalk.yellow(
+        'Potentially conflicting PNPM config detected. Please update your pnpm-workspace.yaml manually with the following config:',
+      ),
+    );
+    console.log(chalk.blue(skuPnpmConfig));
+    console.log(
+      chalk.bold(
+        `After updating your config, please run ${chalk.blue('pnpm install')}.`,
+      ),
+    );
+
+    return;
+  }
+
+  // pnpm-workspace.yaml exists and there is no conflicting config elsewhere, so we can safely
+  // inject the config we want
+  if (action === 'update') {
+    const contents = await readFile(pnpmWorkspaceFilePath, 'utf-8');
+
+    const updatedContents = `${contents}\n${skuPnpmConfig}`;
+    await writeFile(pnpmWorkspaceFilePath, updatedContents, 'utf-8');
+
+    console.log(
+      chalk.bold(
+        `Updated existing pnpm-workspace.yaml. Please run ${chalk.blue('pnpm install')}.`,
+      ),
+    );
+
+    return;
+  }
+};
+
+export const getPnpmConfigAction = async ({
+  npmrcExists,
+  pnpmWorkspaceFileExists,
+}: {
+  npmrcExists: boolean;
+  pnpmWorkspaceFileExists: boolean;
+}): Promise<PnpmConfigAction> => {
+  if (!pnpmWorkspaceFileExists && !npmrcExists) {
+    return 'create';
+  }
+
+  const publicHoistPatterns =
+    (await getPnpmConfigValue('public-hoist-pattern')) ?? [];
+  const onlyBuiltDependencies =
+    (await getPnpmConfigValue('only-built-dependencies')) ?? [];
+
+  const hasExpectedPublicHoistPatterns = skuPublicHoistPatterns.every(
+    (pattern) => publicHoistPatterns.includes(pattern),
+  );
+
+  const hasExpectedOnlyBuiltDependencies = skuOnlyBuiltDependencies.every(
+    (dep) => onlyBuiltDependencies.includes(dep),
+  );
+
+  if (hasExpectedPublicHoistPatterns && hasExpectedOnlyBuiltDependencies) {
+    return 'noop';
+  }
+
+  // It's only safe to update/create the pnpm workspace file if there is no existing config for
+  // these properties. If there is existing config, new config added to pnpm-workspace.yaml will
+  // either override or conflict with it, so we should update and comment the config instead.
+  if (publicHoistPatterns.length === 0 && onlyBuiltDependencies.length === 0) {
+    if (pnpmWorkspaceFileExists) {
+      return 'update';
+    }
+
+    return 'create';
+  }
+
+  // In all other cases, simply log a message instructing the user to update the config manually
+  return 'log';
+};
+
+export const detectPnpmConfigFiles = async (rootDir: string) => {
+  const npmrcExists = await exists(path.join(rootDir, '.npmrc'));
+
+  if (npmrcExists) {
+    console.log(
+      chalk.yellow(
+        "Found '.npmrc' file. It is strongly recommended to migrate all .npmrc configuration to 'pnpm-workspace.yaml' and gitignore '.npmrc'. See https://pnpm.io/settings for more information.",
+      ),
+    );
```

Comment:
suggestion: What do you think about taking our changelogs approach to sentences and having these as separately logged lines?


```suggestion
    console.log(chalk.yellow("Found '.npmrc' file."));
    console.log(chalk.yellow("It is strongly recommended to migrate all .npmrc configurations to 'pnpm-workspace.yaml' and gitignore '.npmrc'. See https://pnpm.io/settings for more information."));
```